ICON_EMBEDDING.md

```diff
@@ -0,0 +1,42 @@
+# Application Icon Embedding
+
+This document describes how the application icon is embedded into Windows executables.
+
+## Overview
+
+The testctxlint Windows executables include an embedded application icon based on the project logo. This provides a better user experience when the executable is displayed in file explorers or task managers.
+
+## Files
+
+- `assets/logo-icon.svg` - Original SVG logo
+- `assets/logo-icon.ico` - Windows ICO format icon (generated from SVG)
+- `cmd/testctxlint/icon.syso` - Windows resource file (compiled from ICO)
+
+## How It Works
+
+1. The SVG logo is converted to ICO format with multiple sizes (16x16, 32x32, 48x48, 64x64, 128x128, 256x256)
+2. The ICO file is compiled into a Windows resource file (.syso) using the `rsrc` tool
+3. The .syso file is automatically included in Windows builds by the Go compiler
+4. The .syso file is ignored on non-Windows builds
+
+## Regenerating the Icon
+
+If the logo changes, you can regenerate the embedded icon:
+
+```bash
+# Convert SVG to ICO (requires ImageMagick)
+convert -background none assets/logo-icon.svg -define icon:auto-resize=256,128,64,48,32,16 assets/logo-icon.ico
+
+# Compile to Windows resource (requires rsrc tool)
+go install github.com/akavel/rsrc@latest
+cd cmd/testctxlint
+rsrc -ico ../../assets/logo-icon.ico -o icon.syso
+```
```

Comment:
Would it be possible to somehow have these instructions embedded into a set of `//go:generate` comments in a new .go file? Just so someone can run `go generate` to achieve the same thing.

Ideally, the `github.com/akavel/rsrc` dependency would be tracked via go.mod and the command instead run via `go run -mod=mod`.